r-package/grf/R/input_utilities.R

```diff
@@ -11,6 +11,55 @@ validate_X <- function(X) {
   if (inherits(X, "Matrix") & !(inherits(X, "dgCMatrix"))) {
       stop("Currently only sparse data of class 'dgCMatrix' is supported.")
   }
+  
+  if (any(is.na(X))){
+    stop("The feature matrix X contains at least one NA.")
+  }
+}
+
+validate_YX <- function(Y,X) {
+  validate_Y(Y)
+  if(length(Y) != nrow(X)) { stop("length(Y) does not equal nrow(X).") }
+}
+
+validate_Y <- function(Y) {
+  if(!is.vector(Y) || !is.numeric(Y)) {
+    stop(paste("The outcome Y must be numeric vector. GRF does not", 
+               "currently support non-numeric or non-vector outcomes."))
+  }
+
+  if (any(is.na(Y))){
+    stop("The outcome vector Y contains at least one NA.")
+  }
+}
+
+validate_WX <- function(W,X) {
+  validate_W(W)
+  if(length(W) != nrow(X)) { stop("length(W) does not equal nrow(X).") }
+}
+
+validate_W <- function(W) {
+  if(!is.vector(W) || !is.numeric(W)) {
+    stop(paste("The treatment W must be numeric vector. GRF does not", 
+               "currently support non-numeric or non-vector treatments."))
+  }
+
+  if(length(unique(W))!=2) {
+    stop(paste("The treatment W must contain only two unique values."))
+  }
+
+  if (any(is.na(W))){
+    stop("The treatment vector W contains at least one NA.")
+  }
+}
+
+validate_ZX <- function(Z,X) {
+  validate_Z(Z)
+  if(length(Z) != nrow(X)) { stop("length(Z) does not equal nrow(X).") }
+}
+
+validate_Z <- function(Z) {
+  TRUE
```

Comment:
Why do we always return `TRUE` when validating the instrument?

Reply:
I didn't find any prior constraints on Z and didn't want to impose the same conditions as those on W and Y.  Judging from the [prior comment](https://github.com/grf-labs/grf/pull/378#discussion_r271132460), it sounds like the Z, W, and Y can all be validated under the same constraints.  I'll follow the suggestion and implement that in the next commits.